### PR TITLE
Generate documentation

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,3 +76,360 @@ be found in [`lib/transformations.js`](lib/transformations.js).
...
+
+`PostgresProvider()`
+
+
+### Create PostgresProvider
+
+`PostgresProvider.prototype.constructor()`
+
+
+### Open PostgresProvider
```
Comment (lundibundi) [inline README.md]: I thought this would be the other way around:

Function/method name as a title
description below

i.e.

-----

### `PostgresProvider.prototype.open(options, callback)`
   // some version descriptions
   - `options: `[`<Object>`] to be passed to pg
   - `callback: `[`<Function>`]
     - `err: `[`<Error>`]` | `[`<null>`]
     - `provider: `[`<this>`] 

Open PostgresProvider

-----
This will give us a meaningful links later on and allow to easily auto-generate the navigation.
I don't think we need description-as-a-title at all, it will only confuse the reader IMO.

File: README.md [hunk 2]
```diff
@@ -76,3 +76,392 @@ be found in [`lib/transformations.js`](lib/transformations.js).
...
+    - `err: `[`<Error>`]` | `[`<null>`]
+
+Link records with Many relation between them
+
+
+
+### `PostgresProvider.prototype.unlinkDetails(category, field, fromId, toIds, callback)`
+  - `category: `[`<string>`] category with field having the Many decorator
+  - `field: `[`<string>`] field with the Many decorator
+  - `fromId` `<Uint64>`Id of the record in category specified in the first
```
Comment (belochub) [inline README.md]: Why is there no space after `` `<Uint64>` ``?

File: README.md (hunk 1 above)
Comment (belochub) [reply README.md]: I don't think function/method name should be a level 3 header.

**Comment by SemenchenkoVitaliy** [comment]
ping @lundibundi @belochub @nechaido.

File: .metadocrc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "lib/fs.provider.js",
+    "lib/memory.provider.js",
+    "lib/pg.provider.js",
+    "lib/remote.provider.js"
+  ],
+  "customLinks": {
+    "pg.Pool": "https://github.com/brianc/node-pg-pool",
+    "pg.Client": "https://github.com/brianc/node-postgres",
+    "Uint64": "https://github.com/metarhia/common#class-uint64",
+    "Metaschema": "https://github.com/metarhia/metaschema/blob/master/lib/schema.js",
```
Comment (SemenchenkoVitaliy) [inline .metadocrc]: If https://github.com/metarhia/metaschema/pull/231 is landed before this PR `Metaschema` link should be replaced with https://github.com/metarhia/metaschema#class-metaschema.

File: README.md [hunk 4]
```diff
@@ -78,3 +78,961 @@ be found in [`lib/transformations.js`](lib/transformations.js).
 ## Distributed Metamodel Repository
 
 ## Data Access Layer
+
+## API
+
+### gs(provider, options)
```
Comment (belochub) [inline README.md]: We need to document the arguments here as well.

File: gs.js [hunk 5]
```diff
@@ -17,7 +17,15 @@ const providers = {
...
+// Create provider
+//   provider <string> provider name
+//   options <Object>
+//     serverSuffix <Uint64> optional
+//     serverBitmask <Uint64> optional
+//     systemSuffix <Uint64> optional
+//     systemBitmas <Uint64> optional
 const gs = (provider, options) => new providers[provider](options);
+
 gs.schemaConfig = require('./lib/metaschema-config/config.js');
```
Comment (belochub) [inline gs.js]: Can we also document this property?

File: doc/header.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+## JavaScript Query Language
+
+JSQL is a query language for data structures manipulation. JSQL have syntax for:
+filter, projection, dataset join and set operations. We have a separate
+repository for examples and specification:
+[metarhia/JSQL](https://github.com/metarhia/JSQL). Current Implementation can
+be found in [`lib/transformations.js`](lib/transformations.js).
+
+## Distributed Metamodel Repository
```
Comment (belochub) [inline doc/header.md]: Can we maybe remove this header and the one below, since there is no text/description on the topics?

File: doc/footer.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+## Contributors
```
Comment (belochub) [inline doc/footer.md]: I think this should be moved to the end of the header file to make it easy to find.

Comment (SemenchenkoVitaliy) [reply doc/footer.md]: In this case, shouldn't we also move it to the end of the header in the other repositories?

File: doc/header.md (hunk 6 above)
Comment (SemenchenkoVitaliy) [reply doc/header.md]: I'd also prefer to remove them since they don't have any useful information, but what about `README.ru.md` which has those headers? Shouldn't they be removed too?

Comment (belochub) [reply doc/header.md]: This PR doesn't change `README.ru.md`, it may be done separately afterward if someone insists on removing it.

File: doc/footer.md (hunk 7 above)
Comment (belochub) [reply doc/footer.md]: We can decide on what to do in other repositories when we modify README files there.

File: doc/header.md [hunk 8]
```diff
@@ -75,8 +75,6 @@ repository for examples and specification:
 [metarhia/JSQL](https://github.com/metarhia/JSQL). Current Implementation can
 be found in [`lib/transformations.js`](lib/transformations.js).
 
-## Distributed Metamodel Repository
+## Contributors
 
-## Data Access Layer
-
-## API
```
Comment (belochub) [inline doc/header.md]: Why did you remove this header?

File: .prettierignore [hunk 9]
```diff
@@ -1,2 +1,3 @@
 data
 dist
+doc
```
Comment (belochub) [inline .prettierignore]: I don't think it should be ignored.

File: README.md [hunk 10]
```diff
@@ -75,6 +75,905 @@ repository for examples and specification:
...
+
+Lazy map
+
+fn - [`<Function>`][function], map function
+
+#### Cursor.prototype.projection(fields)
+
+- `fields`: [`<string[]>`][string]|[`<Object>`][object] projection metadata
+  array of field names or object with structure: { toKey: [ fromKey,
+  functions... ] }
```
Comment (lundibundi) [inline README.md]: Nit: Perhaps wrap '{ toKey: [ fromKey,   functions... ] }' in backticks?
